### ISSUE TITLE
v2.0.7 draws undefined elements when Path Lines enabled

### ISSUE THREAD

**Issue by Serinu** [sghaskell/maps-plus#1]
Hi together,

when using the recent version of Maps+ for Splunk (2.0.7), enabling Path Lines causes an unexpected behavior of the map:
- PathLines will get a new Layer named "undefined" in the LayerControl
- PathLines will get an empty popup on a mouse click
- PathLines will get the tooltip of the first row in the underlying search result

Tested under Splunk 7.2.1 and 6.5.8. Is this a bug? The Legacy Clustered Single Value Map won't create a new layer, popups or tooltips on the Path Lines. Which I assume to be the desired behavior.

```
<dashboard>
  <label>Demo: Bug: Maps+ draws undefined elements when Path Lines enabled</label>
  <row>
    <panel>
      <viz type="leaflet_maps_app.maps-plus">
        <search>
          <query>| makeresults count=4 
| streamstats count AS row
| eval markerVisibility = case(row=1,"marker", row=2,"marker", row=3,"none",  row=4,"marker", 1=1,null())
| eval latitude         = case(row=1,"51.48",  row=2,"51.53",  row=3,"51.55", row=4,"51.58",  1=1,null())
| eval longitude        = case(row=1,"14.10",  row=2,"14.02",  row=3,"13.90", row=4,"14.20",  1=1,null())
| eval description      = "Popup ".row
| eval tooltip          = "Tooltip ".row
| fields - _time row</query>
          <earliest>0</earliest>
          <latest></latest>
        </search>
        <option name="leaflet_maps_app.maps-plus.pathColorList">#0003F0,#D43C29,darkgreen,0xe2d400,darkred,#23A378</option>
        <option name="leaflet_maps_app.maps-plus.showPathLines">1</option>
      </viz>
    </panel>
  </row>
</dashboard>

**Comment by sghaskell** [comment]
Hi @Serinu 

The behavior you describe is the default behavior for paths if you don't specify any additional fields to control them. 

* Use the `description` field to control the mouse click/popup. 
* Use the `layerGroup` field to control the layers. 
* Use the `layerDescription` field to control the label in the layer control legend. 
* Use the `tooltip` field to control the displayed tooltip on the path.

I've got some enhancements that I'm due to release soon. One of those is a `layerIcon` field to control the icon displayed in the layer control legend.

**Comment by cjer** [comment]
I have a related question. Is there a way to disable permanent tooltips for Path Lines, but keep it on for Markers?

**Comment by ericmck2000** [comment]
Going to +1 this issue- Path lines appear as a "Undefined" layer. Cannot determine how to properly set the values.

**Comment by sghaskell** [comment]
This has been addressed and will be released shortly in an upcoming v3.0.0 release. I'll update with details once it has been released.

**Comment by sghaskell** [comment]
This has been addressed in [v3.0.1](https://github.com/sghaskell/maps-plus/releases/tag/3.0.1). See [layer priority](https://github.com/sghaskell/maps-plus#layer-priority) and [layer controls](https://github.com/sghaskell/maps-plus#layer-controls) for details. Paths will get their own layer based on their `id`. Use the `pathLayer` field to group multiple paths together. Use `layerDescription`, `layerIcon`, `layerIconSize`, `layerIconColor` and `layerIconPrefix` to style the layer in the legend.
